### DISCUSSION TITLE
Feature request: Client.read.except_types(*types)

### DISCUSSION THREAD

**Discussion by grncdr** [RailsEventStore/rails_event_store#1212] [Ideas]
We have some streams that are pretty large, and it would be convenient (especially when debugging) to exclude certain event types while reading the stream. I propose a pair of methods like `Specification#except_type` / `#except_types` that do the opposite of `#of_type`.

What do you think? If it makes sense I can try and implement it.

**Comment by mpraglowski** [comment]
Just show me the code ;) Seems that it might have some use.

I'm interested how you debug problems - maybe this is not all we could do. The initial idea for 'of_type' method in #456 was also connected to some "debugging API". We could gather more debug scenarios & implement something that will make debugging (especially on large streams of events) easier.

**Comment by grncdr** [comment]
Well, "debugging" here means:

1. using IRB/pry 
2. using some "admin" views of event streams related to a particular entity (e.g. `/admin/widgets/123/events`)

Especially in the second case, it would be nice to do something like `/admin/widgets/123/events?except_type=SomeEventThatHappensFrequentlyButIsNotImportantRightNow` (or even put some dropdown filtering options on that view 😉).

I don't have any pressing need for a more generalized debugging facility that ships with RailsEventStore itself. I can _imagine_ something like adding a UI for constructing a `Specification` to the existing event store browser, but for my use case it really helps to render the events myself with the additional context available from my app.

**Comment by mostlyobvious** [comment]
Perhaps `except_type` would be useful when re-writing streams — to filter out events we no longer want in new stream.
